server/player/player.go

```diff
@@ -806,6 +806,8 @@ func (p *Player) kill(src damage.Source) {
 		viewer.ViewEntityAction(p, entity.DeathAction{})
 	}
 
+	p.handler().HandleDeath(src)
```

Comment:
I think this should get called after the health is set to zero for the player.

Reply:
Done